odl/space/weighting.py

```diff
@@ -594,6 +603,10 @@ def __eq__(self, other):
         return (super().__eq__(other) and
                 self.array is getattr(other, 'array', None))
 
+    def __hash__(self):
+        """Return ``hash(self)``."""
+        return super().__hash__() ^ hash(self.array)
```

Comment:
`ndarray`'s can't be hashed. But perhaps we can have this anyway to let the error propagate.